caikit_nlp/modules/token_classification/transformer_sentence_classification.py

```diff
@@ -0,0 +1,218 @@
+                        self.labels_to_output and label in self.labels_to_output
+                    ):
+                        sentence_span = sentence_span_list[idx]
+                        token_classification = TokenClassification(
+                            start=sentence_span.start,
+                            end=sentence_span.end,
+                            word=sentence_span.text,
+                            entity=label,
+                            score=classification.score,
+                        )
+                        token_classification_results.append(token_classification)
+        return TokenClassificationResult(results=token_classification_results)
+
+    def save(self, model_path: str):
+        """Save model in target path
+
+        Args:
+            model_path: str
+                Path to store model artifact(s)
+        """
+        module_saver = ModuleSaver(
+            self,
+            model_path=model_path,
+        )
+
+        with module_saver:
+            module_saver.save_module(self.sentence_splitter, "sentence_split")
+            module_saver.save_module(
+                self.sequence_classifier, "sequence_classification"
+            )
+            config_options = {
+                "language": self.lang,
+                "default_threshold": self.default_threshold,
+                "labels_mapping": self.labels_mapping,
+                "labels_to_output": self.labels_to_output,
+            }
+            module_saver.update_config(config_options)
+
+    @classmethod
+    def load(cls, model_path: str) -> "TransformerSentenceClassification":
+        """Load a transformer sentence classification model.
+
+        Args:
+            model_path: str
+                Path to the model to be loaded.
+
+        Returns:
+            TransformerSentenceClassification
+                Instance of this class built from the on disk model.
+        """
+        config = ModuleConfig.load(os.path.abspath(model_path))
+        loader = ModuleLoader(model_path)
+        sentence_splitter = loader.load_module("sentence_split")
+        try:
+            sequence_classifier = loader.load_module("sequence_classification")
+        except Exception:  # pylint: disable=broad-exception-caught
+            # Module loader looks for config.yml so we load this directly,
+            # pending a way to load HF models directly through config.json
+            # - https://github.com/caikit/caikit/issues/236
+            log.info(
+                "<NLP47789919I>",
+                "Model not able to be loaded directly. \
+                    Currently occurs for models without config.yml",
+            )
+            sequence_classification_path = os.path.join(
+                model_path, config.module_paths["sequence_classification"]
+            )
+            sequence_classifier = SequenceClassification.load(
+                sequence_classification_path
+            )
```

Comment:
I think this seems pretty hacky, because:
1. `sentence_split` actually doesn't have a model so its auto-magically getting loaded and doesn't need to be in the `model_path` ? 
2. The trial and error method for this sequence classification looks mixed paradigms. Can we instead following the `bootstrap -> load` paradigm for now until the `caikit.core.model_management.initializer` paradigm gets merged?

Reply:
> sentence_split actually doesn't have a model so its auto-magically getting loaded and doesn't need to be in the model_path ?

Hm there is indeed no concrete implementation is provided here at this time but based on previous feedback (i.e. if I change `SentenceSplitBase` to `ModuleBase`) this line `sentence_splitter = loader.load_module("sentence_split")` will essentially just load a module from the given `model_path`. Only whether or not the loaded sentence split module is directly usable is harder to track at this time

> The trial and error method for this sequence classification looks mixed paradigms.
This is actually fairly intentional for now because the `load_module` works for modules saved with `config.yml`. This is not doing the bootstrapping, and until we have the right direct Huggingface loader to `load_module` this accounts for both cases (model with config.yml and model without [e.g. transformer])

Reply:
Discussed offline - with the move to use bootstrapping (which can happen via resource) and nest the module artifacts for sequence classification one level down with `config.yml`, `load_module` can just be used here